### PR TITLE
Add script to export customers data from releases

### PR DESCRIPTION
**Pull request by KevinTss**
Act as an external of Cycle's team developer, read the README and see if everything is clear enough for you to understand the script and make it run.

One problem I have for now -> For large data it takes time and it fails at some point.

### CODE REVIEW

File: src/examples/read-release-customers/index.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        releaseNoteTitle: releaseNote.title,
+        nbQuotes: insights.length.toString(),
+        nbFeedback: feedbacks.length.toString(),
+        nbCustomer: uniqueCustomers.size.toString(),
+        nbCompany: uniqueCompanies.size.toString(),
+      });
+    });
+  });
+
+  await generateCSV(
```
[the remark this answers is not in the document]
Comment (KevinTss) [reply src/examples/read-release-customers/index.ts]: Fixed, it make the script faster 🔥 Good call

File: src/examples/read-release-customers/index.ts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+          data: ReleaseNoteWithDoc;
+          insights: {
+            [insightId: string]: {
+              data: DocWithSourceDoc;
+            };
+          };
+        };
+      };
+    };
+  } = {};
```
Comment (martin-cycle) [inline src/examples/read-release-customers/index.ts:124]: weird to read, better to define typing somewhere else

File: src/utils/companies.ts [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+        id
+      }
+    }
+  `;
+  const variables = {
+    companyId,
+    ...data,
+  };
+  // console.log('query', query);
+  // console.log('variable', variables);
```
Comment (martin-cycle) [inline src/utils/companies.ts:316]: clean

File: src/utils/companies.ts [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+  const response = await queryCycle<QueryUpdateCompanyResponse>({
+    query,
+    variables,
+  });
+  // console.log('++', response);
```
Comment (martin-cycle) [inline src/utils/companies.ts:321]: clean

Comment (KevinTss) [reply src/utils/companies.ts]: [dde7002](https://github.com/cycle-app/api-examples/pull/1/commits/dde7002e65b3d94ade51a822111f0fdd70ceb6d0)

File: src/examples/read-release-customers/index.ts (hunk 2 above)
Comment (KevinTss) [reply src/examples/read-release-customers/index.ts]: Separated [1c2cf34](https://github.com/cycle-app/api-examples/pull/1/commits/1c2cf348ff3253556e96adea8f68014dabe37004)

File: src/utils/cycle.ts [hunk 5]
```diff
@@ -1,20 +1,46 @@
...
+        const errorText = await response.text();
+        console.error('Error fetching Cycle:', errorText);
+        if (attempt >= MAX_RETRIES) return null;
+      } else {
+        const responseData = await response.json();
+        return responseData;
+      }
+    } catch (error) {
+      console.error(`Attempt ${attempt} failed due to error:`, error);
+      if (attempt >= MAX_RETRIES) return null;
```
Comment (KevinTss) [inline src/utils/cycle.ts:41]:
```suggestion
      if (attempt >= MAX_RETRIES) throw Error("Error fetching cycle")
```

Comment (xuab) [reply src/utils/cycle.ts:41]: I think it's better to throw the original error
```ts
// If cause is supported (node 16.9+)
throw new Error('Error fetching cycle', { cause: error });
// If not supported
throw error;
```

An error should also be thrown if the failure comes from `!response.ok`
```ts
if (!response.ok) {
  const errorText = await response.text();
  console.error('Error fetching Cycle:', errorText);
  if (attempt >= MAX_RETRIES) throw new Error(errorText);
```

Comment (KevinTss) [reply src/utils/cycle.ts]: I did in both since if anything happen in the catch we need the same logic

**Pull request merged**